### PR TITLE
chore(docs): Update the documentation for UpgaredAdapter

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

**Comment by vicb** [comment]
type in the commit comment

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {verifyNoBrowserErrors} from '../../../_common/e2e_util';
+
+function waitForElement(selector: string) {
+  var EC = (<any>protractor).ExpectedConditions;
```
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts]: const

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  // Waits for the element with id 'abc' to be present on the dom.
+  browser.wait(EC.presenceOf($(selector)), 20000);
+}
+
+describe('upgrade', () => {
+  afterEach(verifyNoBrowserErrors);
+
+  describe('downgradeNg2Component', () => {
+    var URL = '/upgrade/downgradeNg2Component/ts/';
```
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts]: const

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import {BrowserModule} from '@angular/platform-browser';
+import {UpgradeAdapter} from '@angular/upgrade';
+import {IAngularStatic} from '@types/angular';
+declare var angular: IAngularStatic;
+
+
+// #docregion downgradeNg2Component
+// Start with instantiating UpgradeAdapter which will be used to create facades between the
+// two framworks.
+var adapter = new UpgradeAdapter(forwardRef(() => MyNg2Module));
```
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: var -> const

File: modules/@angular/upgrade/src/upgrade_adapter.ts [hunk 4]
```diff
@@ -6,94 +6,65 @@
...
 import {ComponentInfo, getComponentInfo} from './metadata';
 import {UpgradeNg1ComponentAdapterBuilder} from './upgrade_ng1_adapter';
 import {controllerKey, onError} from './util';
 
 var upgradeCount: number = 0;
+declare var angular: IAngularStatic;
 
 /**
- * Use `UpgradeAdapter` to allow AngularJS v1 and Angular v2 to coexist in a single application.
+ * @whatItDoes Use `UpgradeAdapter` to allow AngularJS v1 and Angular v2 to coexist in a single
```
Comment (vicb) [inline modules/@angular/upgrade/src/upgrade_adapter.ts]: remove `Use `UpgradeAdapter` to ` ?

File: modules/@angular/upgrade/src/upgrade_adapter.ts [hunk 5]
```diff
@@ -454,67 +341,35 @@ export class UpgradeAdapter {
   }
 
   /**
-   * Allows AngularJS v1 service to be accessible from Angular v2.
+   * @whatItDoes Allows AngularJS v1 service to be accessible from Angular v2 injector.
    *
+   * Cretase a factory in Angular 2 root injector which will retriev an instance from AngularJS v1
```
Comment (vicb) [inline modules/@angular/upgrade/src/upgrade_adapter.ts]: typo `Cretase`, `retriev` seems like your should enable the spell checker :)

**Review by vicb — CHANGES_REQUESTED** [review]
Also CI is red

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts (hunk 1 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts]: fixed

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts (hunk 2 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/downgradeNg2Component_spec.ts]: fixed

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts (hunk 3 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: fixed

File: modules/@angular/upgrade/src/upgrade_adapter.ts (hunk 5 above)
Comment (mhevery) [inline modules/@angular/upgrade/src/upgrade_adapter.ts]: fixed

File: modules/@angular/upgrade/src/upgrade_adapter.ts (hunk 4 above)
Comment (mhevery) [inline modules/@angular/upgrade/src/upgrade_adapter.ts]: fixed

File: modules/@angular/examples/upgrade/overview/ts/module.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+// Place Angular 2 component into its coresponding NgModule
+@NgModule({
+  declarations: [Ng2Component, adapter.upgradeNg1Component('ng1Hello')],
+  imports: [BrowserModule]
+})
+class MyNg2Module {
+}
+
+
+// Crete AngularJS 1 module.
```
Comment (Foxandxss) [inline modules/@angular/examples/upgrade/overview/ts/module.ts]: create

Comment (mhevery) [inline modules/@angular/examples/upgrade/overview/ts/module.ts]: fixed

**Comment by mhevery** [comment]
ping @vicb can I have github review approval?

**Review by vicb — APPROVED** [review]

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts (hunk 3 above)
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: could you fix it again ?

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  @Input() name: string;
+}
+
+// Add the Angular 2 component into NgModule as normally.
+@NgModule({declarations: [Greeter], imports: [BrowserModule]})
+class MyNg2Module {
+}
+
+
+var module = angular.module('myExample', []);
```
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: const

File: modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import {BrowserModule} from '@angular/platform-browser';
+import {UpgradeAdapter} from '@angular/upgrade';
+import {IAngularStatic} from '@types/angular';
+declare var angular: IAngularStatic;
+
+
+// #docregion downgradeNg2Provider
+// Start with instantiating UpgradeAdapter which will be used to create facades between the
+// two framworks.
+var adapter = new UpgradeAdapter(forwardRef(() => MyModule));
```
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts]: const

File: modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+// Also create AngularJS 1 module.
+var module = angular.module('myExample', []);
```
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts]: const

File: modules/@angular/examples/upgrade/overview/ts/module.ts [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+import {BrowserModule} from '@angular/platform-browser';
+import {UpgradeAdapter} from '@angular/upgrade';
+import {IAngularStatic} from '@types/angular';
+declare var angular: IAngularStatic;
+
+
+// #docregion Overview
+
+// Create an UpgradeAdapter which will bridge the two frameworks
+var adapter = new UpgradeAdapter(forwardRef(() => MyNg2Module));
```
Comment (vicb) [inline modules/@angular/examples/upgrade/overview/ts/module.ts]: const

File: modules/@angular/examples/upgrade/overview/ts/module.ts [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+@NgModule({
+  declarations: [Ng2Component, adapter.upgradeNg1Component('ng1Hello')],
+  imports: [BrowserModule]
+})
+class MyNg2Module {
+}
+
+
+// Create AngularJS 1 module.
+var module = angular.module('myExample', []);
```
Comment (vicb) [inline modules/@angular/examples/upgrade/overview/ts/module.ts]: const

File: modules/@angular/examples/upgrade/upgradeNg1Component/ts/module.ts [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+import {BrowserModule} from '@angular/platform-browser';
+import {UpgradeAdapter} from '@angular/upgrade';
+import {IAngularStatic} from '@types/angular';
+declare var angular: IAngularStatic;
+
+
+// #docregion upgradeNg1Componnent
+// Start with instantiating UpgradeAdapter which will be used to create facades between the
+// two framworks.
+var adapter = new UpgradeAdapter(forwardRef(() => MyNg2Module));
```
Comment (vicb) [inline modules/@angular/examples/upgrade/upgradeNg1Component/ts/module.ts]: const * 2

File: modules/@angular/examples/upgrade/upgradeNg1Provider/ts/module.ts [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+import {BrowserModule} from '@angular/platform-browser';
+import {UpgradeAdapter} from '@angular/upgrade';
+import {IAngularStatic} from '@types/angular';
+declare var angular: IAngularStatic;
+
+
+// #docregion upgradeNg1Provider
+// Start with instantiating UpgradeAdapter which will be used to create facades between the
+// two framworks.
+var adapter = new UpgradeAdapter(forwardRef(() => MyModule));
```
Comment (vicb) [inline modules/@angular/examples/upgrade/upgradeNg1Provider/ts/module.ts]: same

File: scripts/ci-lite/offline_compiler_test.sh [hunk 14]
```diff
@@ -11,7 +11,7 @@ PKGS=(
   typescript@2.0.2
   zone.js@0.6.21
   rxjs@5.0.0-beta.11
-  @types/{node@6.0.38,jasmine@2.2.33}
+  @types/{angular@1.5.14,node@6.0.38,jasmine@2.2.33}
```
Comment (vicb) [inline scripts/ci-lite/offline_compiler_test.sh]: q: should this be updated when shrinkwrap is updated ? 

zone, node, jasmine versions seems outdated here ?

**Review by vicb — CHANGES_REQUESTED** [review]
- more changes requested,
- CI is red, does not seem to be a flake

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts (hunk 7 above)
Comment (laco0416) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: `module` is a global variable in node.js context. I feel a little confused about the name...

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts (hunk 3 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: fixed again. :-)

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts (hunk 7 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: don't know what happened fixed again

File: modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts (hunk 8 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts]: don't know what happened fixed again

File: modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts (hunk 9 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/downgradeNg2Provider/ts/module.ts]: don't know what happened fixed again

File: modules/@angular/examples/upgrade/overview/ts/module.ts (hunk 10 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/overview/ts/module.ts]: don't know what happened fixed again

File: modules/@angular/examples/upgrade/overview/ts/module.ts (hunk 11 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/overview/ts/module.ts]: don't know what happened fixed again

File: modules/@angular/examples/upgrade/upgradeNg1Component/ts/module.ts (hunk 12 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/upgradeNg1Component/ts/module.ts]: don't know what happened fixed again

File: modules/@angular/examples/upgrade/upgradeNg1Provider/ts/module.ts (hunk 13 above)
Comment (mhevery) [inline modules/@angular/examples/upgrade/upgradeNg1Provider/ts/module.ts]: don't know what happened fixed again

File: scripts/ci-lite/offline_compiler_test.sh (hunk 14 above)
Comment (mhevery) [inline scripts/ci-lite/offline_compiler_test.sh]: Yes, we should have a better way of doing this, but this is outside of scope of this PR.

**Comment by mhevery** [comment]
@vicb OK, ci is is green. Can you give LGTM

File: modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts (hunk 7 above)
Comment (vicb) [inline modules/@angular/examples/upgrade/downgradeNg2Component/ts/module.ts]: still not here.
@alexeagle is going to have a fix for that.

**Review by vicb — APPROVED** [review]

**Comment by rkirov** [comment]
needs rebasing

**Comment by mhevery** [comment]
@IgorMinar ping...

**Comment by IgorMinar** [comment]
This commit breaks g3 because `@types/angular` is not available. Please take a look at cl/136436863 and prepare a patch.

also please update the commit message type to `docs(upgrade): .. `

**Comment by mhevery** [comment]
Superseded by the static ngUpgrade

**Pull request closed without merging**